hypothesis-python/src/hypothesis/searchstrategy/collections.py

```diff
@@ -144,21 +144,27 @@ def do_draw(self, data):
             max_size=self.max_size,
             average_size=self.average_size,
         )
-        seen = set()
+        seens = tuple(set() for _ in range(len(self.keys)))
```

Comment:
`tuple(set() for _ in self.keys)` is sufficient here, as we're not using the index at all.

I'd also prefer to call the variable `seen_sets`.